### PR TITLE
remove lazy-if-else

### PR DESCRIPTION
**Pull request by uriva**

### CODE REVIEW

**Comment by LawlAoux** [comment]
@uriv Does it work now like lazy if else (from both side)? If yes yes I'm not sure why, because it seems ```if_then_else``` gets all the values, so if the condition is False for example, but true_value raises an exception then it will raise an exception no? Can you also add tests for these? (when true wins and false raises an exception and when false wins and true raises an exception)

**Review by LawlAoux — APPROVED** [review]
@uriv Does it work now like lazy if else (from both side)? If yes yes I'm not sure why, because it seems if_then_else gets all the values, so if the condition is False for example, but true_value raises an exception then it will raise an exception no? Can you also add tests for these? (when true wins and false raises an exception and when false wins and true raises an exception)

**Comment by uriva** [comment]
@LawlAoux the naming is confusing - what's lazy here is not the constituents of the composers but the ambiguity downstream.

In plain english - both `if_true` and `if_false` execute regardless of `condition`.

So why did I write `lazy_if_else` in the first place:
It has two differences from `if_then_else`.
1. it allows for one of the paths to raise.
1. it solves a specific performance problem caused by `first`.

The first feature is largely duplicated with `require`.

As for the second let me explain the problem:

You have two paths going into `first` - `a` and `b`. Let's assume both paths never raise. Then we will compute the entire downstream graph twice, once for the value from `a` and once for the value from `b`. If you have `n` consecutive such compositions, we'll have to compute everything downstream `2^n` times which will quickly become intractable.

Today I realized this is not a problem which is specific to `if_else`, but something general about `first`. So I will need to change the implementation of `first`, to something like this:

```python
def lazy_first_path_which_did_not_raise(*graphs):
    return gamla.pipe(
        graphs,
        enumerate,
        gamla.map(
            gamla.star(
                lambda i, graph: condition.require(
                    composers.make_or(graphs[:i], lambda args: not args), graph
                )
                if i
                else graph
            )
        ),
        gamla.star(composers.make_first),
    )
```

This implementation will ensure the ambiguity doesn't increase.

For now I'm just removing this version because people are misusing it.

**Comment by uriva** [comment]
@LawlAoux the implementation in my previous comment can't work (the raising paths will not enable the lambda to run.
I have to change the cg runner (sent another pr).
